alpha/lib/model/entry.php

```diff
@@ -1951,11 +1951,53 @@ public function getEntitledKusersEditArray()
 		return array_keys($this->getEntitledUserPuserEditArray());
 	}
 	
+	public function getEntitledKusersView()
+	{
+		return implode(',', array_keys($this->getEntitledPusersViewArray()));
+	}
+
+	public function getEntitledKusersViewArray()
+	{
+		return array_keys($this->getEntitledPusersViewArray());
+	}
+	
 	public function getEntitledPusersEdit()
 	{
 		return implode(',', $this->getEntitledUserPuserEditArray());
 	}
 	
+	public function setEntitledPusersView($v)
+	{
+		$entitledUserPuserView = array();
+		
+		$v = trim($v);
+		if($v == '')
```

Comment:
This block is not really needed , since a case of empty array will result the same in the foreach loop below.

Reply:
@MosheMaorKaltura  this block deals with the case of setting the string to ''. It is needed.